Helper/Config.php

```diff
@@ -250,6 +250,11 @@ class Config extends AbstractHelper
         self::SUCCESS_PAGE_ACTION
     ];
 
+    public static $supportableProductTypes = [
+        \Magento\Catalog\Model\Product\Type::TYPE_SIMPLE,
+        \Magento\Catalog\Model\Product\Type::TYPE_VIRTUAL,
+        \Magento\Downloadable\Model\Product\Type::TYPE_DOWNLOADABLE,
```

Comment:
downloadable product requires login to purchase. I'm ok not supporting it if it makes logic simpler.

Reply:
After thinking it through I agree with @daisy1754, removing it makes more sense.